src/node.ts

```diff
@@ -89,7 +91,7 @@ function compileHook(code: string, filename: string) {
 
 function hookExtensions(exts: readonly string[]) {
   if (piratesRevert) piratesRevert();
-  piratesRevert = addHook(compileHook, { exts: exts as string[], ignoreNodeModules: true });
+  piratesRevert = addHook(compileHook, { exts: exts as string[], ignoreNodeModules: true, matcher: shouldIgnoreFactory(transformOpts.ignore, transformOpts.only) });
```

Comment:
> I think unit tests at https://github.com/swc-project/swc/blob/69123ddc283f55b224c9bee218065926cba5ae64/.github/workflows/integration.yml#L84-L116 can be ported?

@kdy1 

Babel has its own `ignore` and `only` supports, so their implementation can pass the configuration directly.
`@swc/core` doesn't have `only` and `ignore`, so this PR is based on `pirates`'s `matcher` option.

However, in current unit test cases, the package `pirates` has been hijacked, thus the behavior can not be tested:

https://github.com/swc-project/register/blob/377bad1f81cc590b95cb207c0524e95718b4e065/__tests__/index.js#L12

Reply:
I can't remember the reason it's mocked.
Will removing the mock break other tests?

Reply:
> I can't remember the reason it's mocked. Will removing the mock break other tests?

Yes, removing the `mock` breaks the test. I will try to work around it.

Reply:
I just switch back to the `master` branch and the test still failed.

And I notice that the test cases are ported from `@babel/register`, which has been changed a lot since then.

Reply:
Yeah, some node pakcages of swc need more work.
There was lots of contributions to cli, but not others and I didn't have time to manage all packages.